src/messaging/tests/TestMessagingLayer.h

```diff
@@ -0,0 +1,35 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file declares test entry points for CHIP Messaging layer
+ *      layer library unit tests.
+ *
+ */
+
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
```

Comment:
Why extern "C" here ? The `TestExchangeMgr` function is not defined as C function but C++ function

Reply:
This is a convention used throughout our codebase, see inet or transport. Since nlunit_test is written in C, treat TestExchangeMgr as a C function make more sense to me. 

int TestExchangeMgr()
{
    // Run test suit against one context
    nlTestRunner(&sSuite, &sContext);

    return (nlTestRunnerStats(&sSuite));
}

But I also think removing  extern "C" also works since we have not pass this function name back to nlunit_test.